Extensions/Guardrail+Settings.swift

```diff
@@ -9,8 +9,76 @@
 import HealthKit
 
 public extension Guardrail where Value == HKQuantity {
-    static let correctionRange = Guardrail(absoluteBounds: 60...180, recommendedBounds: 70...120, unit: .milligramsPerDeciliter)
+    static let suspendThreshold = Guardrail(absoluteBounds: 67...110, recommendedBounds: 74...80, unit: .milligramsPerDeciliter)
 
+    static func maxSuspendThresholdValue(correctionRangeSchedule: GlucoseRangeSchedule?, preMealTargetRange: DoubleRange?, workoutTargetRange: DoubleRange?, unit: HKUnit) -> HKQuantity {
+
+        return [
+            suspendThreshold.absoluteBounds.upperBound.doubleValue(for: unit),
+            correctionRangeSchedule?.minLowerBound().doubleValue(for: unit),
+            preMealTargetRange?.minValue,
+            workoutTargetRange?.minValue
+        ]
+        .compactMap { $0 }
+        .min()
+        .map { HKQuantity(unit: unit, doubleValue: $0) }!
+    }
+
+    static let correctionRange = Guardrail(absoluteBounds: 87...180, recommendedBounds: 101...115, unit: .milligramsPerDeciliter)
+
+    static func minCorrectionRangeValue(suspendThreshold: GlucoseThreshold?, unit: HKUnit) -> HKQuantity {
+        return [
+            correctionRange.absoluteBounds.lowerBound.doubleValue(for: unit),
+            suspendThreshold?.value
```

Comment:
you are making an assumption here that the value of the suspend threshold matches the units passed in.